packages/x-tree-view/src/internals/plugins/useTreeViewJSXNodes/useTreeViewJSXNodes.tsx

```diff
@@ -34,20 +38,41 @@ export const useTreeViewJSXNodes: TreeViewPlugin<UseTreeViewJSXNodesSignature> =
         ...prevState,
         nodes: {
           ...prevState.nodes,
-          nodeMap: { ...prevState.nodes.nodeMap, [node.id]: node },
+          nodeMap: { ...prevState.nodes.nodeMap, [node.id]: { ...node, index: -1 } },
           // For `SimpleTreeView`, we don't have a proper `item` object, so we create a very basic one.
           itemMap: { ...prevState.nodes.itemMap, [node.id]: { id: node.id, label: node.label } },
         },
       };
     });
   });
 
+  const setJSXItemsChildrenIndexes = (
+    parentId: string | null,
+    indexes: { [id: string]: number },
+  ) => {
+    setState((prevState) => ({
+      ...prevState,
+      nodes: {
+        ...prevState.nodes,
+        itemIndexes: {
+          ...prevState.nodes.itemIndexes,
+          [parentId ?? TREE_VIEW_ROOT_PARENT_ID]: indexes,
```

Comment:
Nitpick: Have you considered prefixing the constant with `DEFAULT_` as it looks like it is its purpose? 🤔

Reply:
I don't see it as a default but more an ID for the root items when it's needed.
It's not "I don't have parent, let's use this ID", but more "I am a root element, so my parent ID is the root ID".

It's the equivalent to `GRID_ROOT_GROUP_ID` on the grid.